### ISSUE TITLE
Integration on linux with "--enable-test-discovery"

### ISSUE THREAD

**Issue by patrick-zippenfenig** [allegro/swift-junit#11]
Hi,

is there a way to use it with "--enable-test-discovery"?

As tests for linux are now automatically synthesised there is no LinuxMain.swift anymore.

Best, Patrick

**Comment by ar4s** [comment]
When you build on linux environment you have to add test observer (`SwiftTestReporter.TestObserver`) and `LinuxMain.swift` is the best place for that.
Alternatively, you can generate `LinuxMain.swift` by following command `swift test --generate-linuxmain --enable-test-discovery` and then add observer.

**Comment by patrick-zippenfenig** [comment]
Yes, this is basically using the old `LinuxMain.swift` approach. `--generate-linuxmain` will always generate a new LinuxMain.swift and remove the observer after every execution. With bash scripting the observer could be added afterwards, but this seems to be an ugly solution.

For a CI/CD integration that automatically adds news tests, I was hoping that there are other options to enable the observer without having to use `LinuxMain.swift`.

**Comment by aufflick** [comment]
A hack I'm using is to create a test file `AATests.swift` with a no-op test that calls the initialization code, ie:

```swift
import XCTest
import SwiftTestReporter

class AATests: XCTestCase {
  func testAA() {
    _ = TestObserver()
  }
}
```

This works because only tests that start after the initialization will be captured, and (currently) spm evaluates tests in alphabetical order. If test order randomization happens one day this technique wouldn't work with that.

Another hack I thought of would be to make an Objective-C class that implements `+ load` or `+ initialize` and depend on that as a base class or something.
